src/components/annotations/draw.js

```diff
@@ -118,6 +118,19 @@ function drawOne(gd, index) {
         .call(Color.stroke, options.bordercolor)
         .call(Color.fill, options.bgcolor);
 
+    var isSizeConstrained = options.width || options.height;
+
+    var annClipID = 'clip' + fullLayout._uid + '_ann' + index;
+    var annTextClip = fullLayout._defs.select('.clips')
+        .selectAll('#' + annClipID)
+        .data(isSizeConstrained ? [0] : []);
+
+    annTextClip.enter().append('clipPath')
+        .classed('annclip', true)
+        .attr('id', annClipID)
+      .append('rect');
+    annTextClip.exit().remove();
```

Comment:
Ha I see. We can revisit this when we shift to a more d3-idiomatic pattern in annotation `draw`. Then, we could easily remove `<clipPath>` inside the `exit()` selection similar to how updatemenu buttons are [removed](https://github.com/plotly/plotly.js/blob/ce06ea9cea377c5ed87ed7baa088a14a1997dbb7/src/components/updatemenus/draw.js#L103).